plugins/backend/backstage-plugin-argo-cd-backend/src/service/ArgoCdBuilder.ts

```diff
@@ -0,0 +1,461 @@
+import Router from 'express-promise-router';
+import express from 'express';
+import { Config } from '@backstage/config';
+import { LoggerService } from '@backstage/backend-plugin-api';
+import { ArgoService } from './argocd.service';
+import { errorHandler } from '@backstage/backend-common';
+import { getArgoConfigByInstanceName } from '../utils/getArgoConfig';
+import { Logger } from 'winston';
+
+export type ArgoCDBuilderReturn = Promise<{
+  router: express.Router;
+}>;
+
+export interface ArgoCDEnvironment {
+  logger: LoggerService;
+  config: Config;
+}
+
+export class ArgoCDBuilder {
```

Comment:
Is this builder class needed? Could we reuse `createRouter` from `src/routes.ts` directly in `src/plugin.ts`?